### PR TITLE
test/integration: Test more conflict resolution cases

### PR DESCRIPTION
**Pull request by sebn**
Extracted from https://github.com/cozy-labs/cozy-desktop/pull/1334.
Needs https://github.com/cozy-labs/cozy-desktop/pull/1414.

Merging local dir move then remote dir addition to the same
destination doesn't trigger a conflict although it should.

This will be fixed later.

- [x] PR is not too big
- [x] it improves UX & DX in some way
- [x] it includes tests matching the implementation changes
- [ ] it includes relevant documentation

### CODE REVIEW

File: test/integration/conflict_resolution.js [hunk 1]
```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
...
+    // Simulate client restart (always starting from local)
+    await helpers.local.scan()
+    await helpers.remote.pullChanges()
+    await helpers.syncAll()
+  }
+
+  const bothSides = (tree) => ({local: tree, remote: tree})
+
+  describe('merging local file then remote one', () => {
+    it('renames the remote file', async () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: We actually don't know which one is renamed with this test. Or did I miss something?

File: test/integration/conflict_resolution.js [hunk 2]
```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
...
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'same-name-conflict-...',
+        'same-name/'
+      ]))
+    })
+  })
+
+  describe('merging local file addition then remote file move to the same destination', () => {
+    it('renames the remote move destination', async () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: Same here.

File: test/integration/conflict_resolution.js [hunk 3]
```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
...
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'dst',
+        'dst-conflict-...'
+      ]))
+    })
+  })
+
+  describe('merging local file move then remote file addition', () => {
+    it('renames the remote file', async () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: Same here.

File: test/integration/conflict_resolution.js [hunk 4]
```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
...
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'dst',
+        'dst-conflict-...'
+      ]))
+    })
+  })
+
+  describe('merging local dir addition then remote move to the same destination', () => {
+    it('renames the remote move destination', async () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: Same here.

File: test/integration/conflict_resolution.js [hunk 5]
```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
...
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'dst-conflict-.../',
+        'dst/'
+      ]))
+    })
+  })
+
+  describe('merging local dir move then remote dir addition to the same destination', () => {
+    it('renames the remote dir', async () => {
```
Comment (taratatach) [inline test/integration/conflict_resolution.js]: Should we really merge this test when the resulting situation is not the one we're actually expecting?

Comment (taratatach) [reply test/integration/conflict_resolution.js]: I don't know, I have the feeling this does the opposite. Maybe I'm wrong though.

Comment (sebn) [reply test/integration/conflict_resolution.js]: Ok, I'll replace it with a notice and move it with the fix.

File: test/integration/conflict_resolution.js (hunk 1 above)
Comment (sebn) [reply test/integration/conflict_resolution.js]: You're right, one *may* know it actually works this way, but the test definitely not shows anything.

I'll change the test description to something like *it renames one of the files*.

File: test/integration/conflict_resolution.js (hunk 2 above)
Comment (sebn) [reply test/integration/conflict_resolution.js]: Same as [above](https://github.com/cozy-labs/cozy-desktop/pull/1415#discussion_r249010020)

File: test/integration/conflict_resolution.js (hunk 3 above)
Comment (sebn) [reply test/integration/conflict_resolution.js]: Same as [above](https://github.com/cozy-labs/cozy-desktop/pull/1415#discussion_r249010020)

File: test/integration/conflict_resolution.js [hunk 6]
```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
...
+  })
+
+  describe('merging local dir addition then remote move to the same destination', () => {
+    it('renames the remote move destination', async () => {
+      await helpers.local.syncDir.ensureDir('src')
+      await helpers.local.scan()
+      await helpers.syncAll()
+      // FIXME: Initial tree helper?
+      const remoteFile = await cozy.files.statByPath(`/src`)
+      await cozy.files.updateAttributesById(remoteFile._id, {name: 'dst'})
```
Comment (sebn) [inline test/integration/conflict_resolution.js]: s/remoteFile/remoteDir/g

File: test/integration/conflict_resolution.js (hunk 4 above)
Comment (sebn) [reply test/integration/conflict_resolution.js]: Also added *dir* in *remote dir move*.

**Comment by sebn** [comment]
Rebased on master & fixed up commit according to the feedback.

**Comment by sebn** [comment]
Rebased on master / 3.12.0

**Pull request merged**